### PR TITLE
Remove special PyMap, PyFlatMap, etc.

### PR DESCRIPTION
**Pull request by robertwb**
Prefer to use generic MapToFields instead.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
This depends on https://github.com/apache/beam/pull/28462

File: sdks/python/apache_beam/yaml/yaml_transform_unit_test.py [hunk 1]
```diff
@@ -950,43 +948,6 @@ def test_ensure_errors_consumed_no_output_in_error_handling(self):
...
-  def test_expand_pipeline_with_spec(self):
-    with new_pipeline() as p:
-      spec = '''
-        pipeline:
-          type: chain
-          transforms:
-            - type: Create
-              config:
-                elements: [1,2,3]
-            - type: PyMap
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/yaml/yaml_transform_unit_test.py:984]: These are more change-detector tests, they don't really test anything useful.

File: sdks/python/apache_beam/yaml/yaml_provider.py [hunk 2]
```diff
@@ -508,19 +510,14 @@ def _parse_window_spec(spec):
...
-              python_callable.PythonCallableWithSource(fn)),
-          'PyMapTuple': lambda fn: beam.MapTuple(
-              python_callable.PythonCallableWithSource(fn)),
-          'PyFlatMap': lambda fn: beam.FlatMap(
-              python_callable.PythonCallableWithSource(fn)),
-          'PyFlatMapTuple': lambda fn: beam.FlatMapTuple(
-              python_callable.PythonCallableWithSource(fn)),
-          'PyFilter': lambda keep: beam.Filter(
-              python_callable.PythonCallableWithSource(keep)),
+          'LogForTesting': lambda: beam.Map(log_and_return),
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/yaml/yaml_provider.py]: Logging every element is certainly not to be encouraged for production pipelines...I might go so far as to consider it an anti-pattern. But I could see it being used to log bad elements or something similar. We can always add it later.

**Comment by robertwb** [comment]
Run Python_Runners PreCommit

File: sdks/python/apache_beam/yaml/yaml_provider.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (damccorm) [reply sdks/python/apache_beam/yaml/yaml_provider.py]: Agreed on all fronts, the log on a certain branch path does seem useful to me. I also kinda think that if you use something named "Log" you'll know what you're getting, and naming it "LogForTesting" doesn't really discourage use. I'm fine deferring this though

**Review by damccorm — APPROVED** [review]

**Pull request merged**